lib/vfs.js

```diff
@@ -1,65 +1,93 @@
+"use strict";
+
 var fs = require("fs");
+var utils = require("./utils");
+
+function ensureDirectory(path) {
+    if ("/" !== path[path.length - 1]) {
+        path += "/";
+    }
+
+    return path;
+}
 
 module.exports = function vfs(request) {
     return {
         getFile: function getFile(path, cb) {
-            request("/api/vfs/" + path, function(err, response, body) {
-                if (err) return cb(err);
+            var url = "/api/vfs/" + path;
+            var action = "getting file with path " + path;
+
+            request(url, function (err, response, body) {
+                err = err || utils.resolveHttpError(response, "Error " + action + ".");
```

Comment:
No, because it needs to send the contents of the file back as the primary argument. And since it's not JSON, the other callback factory doesn't fit either.